src/clarity-angular/popover/signpost/signpost-content.ts

```diff
@@ -0,0 +1,83 @@
+import {
+    Component, ElementRef, HostListener
+    // AfterViewInit, HostListener, OnDestroy, OnInit, Renderer2, ElementRef
+} from "@angular/core";
+import { Signpost } from "./signpost";
+
+@Component({
+    selector: "clr-signpost-content",
+    template: `
+        <div class="signpost-flex-wrap">
+            <div class="popover-pointer"></div>
+            <div class="signpost-content-header">
+                <button type="button" class="signpost-action close" aria-label="Close" (click)="close()">
+                    <clr-icon aria-hidden="true" shape="close"></clr-icon>
+                </button>
+            </div>
+            <div class="signpost-content-body">
+                <ng-content></ng-content>
+            </div>
+        </div>
+    `,
+    host: {
+        "[class.signpost-content]": "true"
+    }
+})
+export class SignpostContent/* implements OnDestroy, AfterViewInit*/ {
+
+    /**********
+     * @param nbClick
+     *
+     * @description
+     * Used to keep count of the clicks heard by the @HostListener. We always, always, always ignores the first click
+     * (i.e - the click that opened the SignpostContent)
+     * @type {number}
+     */
+    private nbClick: number = 0;
+
+    /**********
+     * @function close
+     *
+     * @description
+     * Close function that uses the signpost instance to toggle the state of the content popover.
+     *
+     */
+    close() {
+        // This can only be called when a signpost content is open
+        this.signpost.toggleSignpost();
+    }
+
+    constructor( private signpost: Signpost, private elementRef: ElementRef ) { }
```

Comment:
You probably want to inject directly the `IfOpenService` rather than use the one from the Signpost parent component. I'm pretty sure you don't even need the parent component anymore.